File: RosenW-DaliPeche/node-koa/src/asserts/exceptions.js
class CustomError extends Error {
  constructor (err) {
    super(err);
  }
}
class UserError extends CustomError {
  constructor (err, statusCode) {
    super(err);
    this.statusCode = statusCode;
  }
}

class PeerError extends CustomError {
  constructor (err, statusCode) {
    super(err);
    this.statusCode = statusCode;
  }
}

class AppError extends CustomError {
  constructor (err, statusCode) {
    super(err);
    this.statusCode = statusCode;
  }
}

module.exports = {
  UserError,
  PeerError,
  AppError,
};

File: RosenW-WebServer/v2/error/exceptions.py
class PeerError(Exception):
  def __init__(self, message, status_code):
    super().__init__(message)
    self.status_code = status_code

class UserError(Exception):
  def __init__(self, message, status_code):
    super().__init__(message)
    self.status_code = status_code